utils.cpp

```diff
@@ -344,19 +353,44 @@ int load_file_v_1(const char str[], Type *** mem, int n_rows, int n_columns, lon
       return -1;
     }
 
-    res = fread ((*mem)[i], sizeof(Type), n_columns, file);
-    //res = fread ((*mem + i), sizeof(Type), n_columns, file);
-    if (res < n_columns) {
-      fprintf (stderr, "Error: fread < 0 when reading file %s\n", str);
-      return -1;
+    // Read the requested columns into mem
+    if (word_length == -1) {
+      res = fread ((*mem)[i], sizeof(Type), n_columns, file);
+      if (res < n_columns) {
+        fprintf (stderr, "Error: fread < 0 when reading file %s\n", str);
+        return -1;
+      }
+    } else {
+      for (int col = 0; col < n_columns; col++) {
+        int sample = 0;
+        for (int b = 0; b < word_length; b++) {
+          if (offset_bit > 7) {
+            current_byte = fgetc(file);
+            offset_bit = 0;
+          }
+          int bit = ((current_byte >> (7-offset_bit)) & 1);
+          if (bit == 1)
+            sample |= (1 << b);
+          offset_bit++;
+        }
+        (*mem)[i][col] = (Type) sample;
```

Comment:
I think in some weird cases (trace_type=d, word_length=...) the sample will be 64b.
So "int" is too small for sample.